src/simulator/cube.py

```diff
@@ -23,28 +23,37 @@ def bounding_box(self, position: Position) -> Box:
         l = self.length
         w = self.width
         h = self.height
+
         # Because the cube has symmetry, we only need the top side points of the disk.
         # We calculate the distance of the farthest points and then get the dimensions
         # of the bounding box. Note, since we used the top points, because of symmetry,
         # we double the dimensions of the bounding box so that the box surrounds the bottom
         # points of the disk.
         points = [[l / 2, w / 2, h / 2], [-l / 2, w / 2, h / 2],
                   [l / 2, -w / 2, h / 2], [-l / 2, -w / 2, h / 2]]
-        new_x = list()
-        new_y = list()
-        new_z = list()
+
+        # like [x-points, y-points, z-points]
+        new_points = [[], [], []]
```

Comment:
have you noticed that the rest of the function is identical for Cylinder and Cube? Maybe you can just take it to Shape and ask each object to give you its extremity_points. These seem to be the only difference. 
I don't know if this will be the same for all our future shapes. But for now your life becomes easier by avoiding duplicated code.

Reply:
It is similar. It can be put in the shape class, only for other shapes that are going to be added (for example, a cone, maybe :/), the code must be changed, it is not just a point (because it is asymmetric). what do you think?

Reply:
Hmmm. Okay, good point. better to leave it as it is. thanks.